### PR TITLE
[SP-2841] - Backport of PDI-14403 - Salesforce: Test Connection shows…

### PR DESCRIPTION
**Pull request by YuryBY**
… variable, instead of real value (6.1 Suite)

Show real username when testing the Salesforce connection

@mchen-len-son, @pamval, here is backport of https://github.com/pentaho/pentaho-kettle/pull/2257/files to 6.1. Thanks.

### CODE REVIEW

File: plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceinput/SalesforceInputDialog.java [hunk 1]
```diff
@@ -1323,13 +1323,14 @@ private void test() {
...
+    String realUsername = null;
     try {
       SalesforceInputMeta meta = new SalesforceInputMeta();
       getInfo( meta );
 
       // get real values
       String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
-      String realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
       String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```
Comment (mbatchelor) [inline plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceinput/SalesforceInputDialog.java]: See comment above for this. Work this with Yury Bakhmutski

File: plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceinsert/SalesforceInsertDialog.java [hunk 2]
```diff
@@ -702,34 +702,53 @@ public boolean tableItemInserted( TableItem tableItem, ValueMetaInterface v ) {
...
+
       // check if the user is given
       if ( !checkUser() ) {
         return;
       }
 
-      connection = getConnection();
+      String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```
Comment (mbatchelor) [inline plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceinsert/SalesforceInsertDialog.java]: See above about PDI-15652

File: plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceupdate/SalesforceUpdateDialog.java [hunk 3]
```diff
@@ -662,9 +663,11 @@ private void test() {
         return;
       }
 
+      String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```
Comment (mbatchelor) [inline plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceupdate/SalesforceUpdateDialog.java]: See above about PDI-15652

File: plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceupsert/SalesforceUpsertDialog.java [hunk 4]
```diff
@@ -747,9 +748,10 @@ private void test() {
         return;
       }
 
-      connection =
-        new SalesforceConnection( log, transMeta.environmentSubstitute( meta.getTargetURL() ), transMeta
-          .environmentSubstitute( meta.getUserName() ), transMeta.environmentSubstitute( meta.getPassword() ) );
+      String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```
Comment (mbatchelor) [inline plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforceupsert/SalesforceUpsertDialog.java]: See above about PDI-15652

File: plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforcedelete/SalesforceDeleteDialog.java [hunk 5]
```diff
@@ -548,9 +549,11 @@ private void test() {
         return;
       }
 
+      String realURL = transMeta.environmentSubstitute( meta.getTargetURL() );
+      realUsername = transMeta.environmentSubstitute( meta.getUserName() );
+      String realPassword = transMeta.environmentSubstitute( meta.getPassword() );
```
Comment (mbatchelor) [inline plugins/salesforce/src/org/pentaho/di/ui/trans/steps/salesforcedelete/SalesforceDeleteDialog.java]: For this line, you need to use Encr.decryptPasswordOptionallyEncrypted( transMeta.environmentSubstitute( xxxx ) ) . See PDI-15652.

**Comment by YuryBY** [comment]
@mbatchelor, the PR is updated taking into account https://github.com/pentaho/pentaho-kettle/commit/1207d12107f709209f14fe7abfdde7d1b134dff8. Thanks.

**Pull request merged**